Failure to build component containing flags shape

I am using cargo-component from commit 29fea4af8a28857e6a1fee3df3cd930a3fadad10.

While trying to build a component that contains a `flags` shape (example [here](https://github.com/bytecodealliance/cargo-component/blob/dab8fe09e1f43f7b47aaaa40b275078864f37a7d/example/wit/example.wit)), the following error is presented (reproduced [here](https://github.com/bytecodealliance/cargo-component/actions/runs/5908361401/job/16027700973?pr=118)):
> error[E0433]: failed to resolve: use of undeclared crate or module `wit_bindgen`
 --> src/lib.rs:1:1
  |
1 | cargo_component_bindings::generate!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `wit_bindgen`
  |
  = note: this error originates in the macro `cargo_component_bindings::generate` (in Nightly builds, run with -Z macro-backtrace for more info)
error[E0412]: cannot find type `Permissions` in this scope
 --> src/lib.rs:1:1
  |
1 | cargo_component_bindings::generate!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
  |
  = help: consider importing this struct:
          std::fs::Permissions
  = note: this error originates in the macro `cargo_component_bindings::generate` (in Nightly builds, run with -Z macro-backtrace for more info)
error[E0433]: failed to resolve: use of undeclared type `Permissions`
 --> src/lib.rs:1:1
  |
1 | cargo_component_bindings::generate!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of undeclared type `Permissions`
  |
  = help: consider importing this struct:
          std::fs::Permissions
  = note: this error originates in the macro `cargo_component_bindings::generate` (in Nightly builds, run with -Z macro-backtrace for more info)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,7 +70,7 @@ tokio = { version = "1.32.0", default-features = false, features = [
 tokio-util = "0.7.8"
 heck = "0.4.1"
 semver = "1.0.18"
-serde = { version = "1.0.183", features = ["derive"] }
+serde = { version = "1.0.185", features = ["derive"] }
 serde_json = "1.0.105"
 indexmap = "2.0.0"
 url = { version = "2.4.0", features = ["serde"] }
@@ -91,7 +91,9 @@ bytes = "1.4.0"
 proc-macro2 = "1.0.66"
 quote = "1.0.33"
 syn = "2.0.29"
-wit-bindgen-rust-lib = "0.10.0"
-wit-bindgen-core = "0.10.0"
-wit-bindgen-rust = "0.10.0"
-wit-bindgen = "0.10.0"
+
+# TODO: Update on next release to not use git
+wit-bindgen-rust-lib = { git = "https://github.com/bytecodealliance/wit-bindgen" }
+wit-bindgen-core = { git = "https://github.com/bytecodealliance/wit-bindgen" }
+wit-bindgen-rust = { git = "https://github.com/bytecodealliance/wit-bindgen" }
+wit-bindgen = { git = "https://github.com/bytecodealliance/wit-bindgen" }

--- a/crates/bindings/src/lib.rs
+++ b/crates/bindings/src/lib.rs
@@ -8,3 +8,7 @@ pub use cargo_component_macro::generate;
 // Re-export `wit_bindgen::rt` module for the generated code to use.
 #[doc(hidden)]
 pub use wit_bindgen::rt;
+
+// Re-export `wit_bindgen::bitflags` module for the generated code to use.
+#[doc(hidden)]
+pub use wit_bindgen::bitflags;

--- a/crates/macro/src/lib.rs
+++ b/crates/macro/src/lib.rs
@@ -307,8 +307,8 @@ impl Config {
         let opts = Opts {
             exports,
             ownership: self.ownership,
-            runtime_path: Some("cargo_component_bindings::rt".to_string()),
-            bitflags_path: Some("cargo_component_bindings::bitflags".to_string()),
+            runtime_path: Some("::cargo_component_bindings::rt".to_string()),
+            bitflags_path: Some("::cargo_component_bindings::bitflags".to_string()),
             ..Default::default()
         };
 

--- a/crates/macro/src/lib.rs
+++ b/crates/macro/src/lib.rs
@@ -308,6 +308,7 @@ impl Config {
             exports,
             ownership: self.ownership,
             runtime_path: Some("cargo_component_bindings::rt".to_string()),
+            bitflags_path: Some("cargo_component_bindings::bitflags".to_string()),
             ..Default::default()
         };
 

--- a/tests/build.rs
+++ b/tests/build.rs
@@ -367,7 +367,11 @@ fn empty_world_with_dep_valid() -> Result<()> {
             package foo:bar
 
             world the-world {
-                export hello: func()
+                flags foo {
+                    bar
+                }
+        
+                export hello: func() -> foo
             }
         ",
     )?;
@@ -376,12 +380,12 @@ fn empty_world_with_dep_valid() -> Result<()> {
         project.root().join("src/lib.rs"),
         "
             cargo_component_bindings::generate!();
-            use bindings::TheWorld;
+            use bindings::{TheWorld, Foo};
             struct Component;
 
             impl TheWorld for Component {
-                fn hello() {
-                    todo!()
+                fn hello() -> Foo {
+                    Foo::BAR
                 }
             }
         ",

